app/connectors/CodeListConnector.scala

```diff
@@ -73,25 +86,25 @@ class FileBasedCodeListConnector @Inject()(appConfig: AppConfig) extends CodeLis
     (codeItem: CodeItem, locale: Locale) => ProcedureCode(codeItem.code, codeItem.getDescriptionByLocale(locale))
   )
 
-  private val additionalProcedureCodeListsByLang = readCodesFromFile(
+  private val additionalProcedureCodeMapsByLang = readCodesMapFromFile(
     appConfig.additionalProcedureCodes,
     (codeItem: CodeItem, locale: Locale) => AdditionalProcedureCode(codeItem.code, codeItem.getDescriptionByLocale(locale))
   )
 
-  private val additionalProcedureCodeForC21ListsByLang = readCodesFromFile(
+  private val additionalProcedureCodeForC21MapsByLang = readCodesMapFromFile(
     appConfig.additionalProcedureCodesForC21,
     (codeItem: CodeItem, locale: Locale) => AdditionalProcedureCode(codeItem.code, codeItem.getDescriptionByLocale(locale))
   )
 
   def getProcedureCodes(locale: Locale): Seq[ProcedureCode] =
-    procedureCodeListsByLang.getOrElse(locale, procedureCodeListsByLang.value.head._2)
+    procedureCodeListsByLang.getOrElse(locale.getLanguage, procedureCodeListsByLang.value.head._2)
 
   def getProcedureCodesForC21(locale: Locale): Seq[ProcedureCode] =
-    procedureCodeForC21ListsByLang.getOrElse(locale, procedureCodeForC21ListsByLang.value.head._2)
+    procedureCodeForC21ListsByLang.getOrElse(locale.getLanguage, procedureCodeForC21ListsByLang.value.head._2)
 
-  def getAdditionalProcedureCodes(locale: Locale): Seq[AdditionalProcedureCode] =
-    additionalProcedureCodeListsByLang.getOrElse(locale, additionalProcedureCodeListsByLang.value.head._2)
+  def getAdditionalProcedureCodesMap(locale: Locale): Map[String, AdditionalProcedureCode] =
+    additionalProcedureCodeMapsByLang.getOrElse(locale.getLanguage, additionalProcedureCodeMapsByLang.value.head._2)
```

Comment:
The `def` could potentially return not what you expect if the provided `locale` is not found. I think we should return the English map in that case and being this a map we cannot be sure the `head` is the English one and not the Welsh for instance. Am I wrong?

Reply:
Yes good spot. Ok altered readCodesMapFromFile to return a ListMap to ensure the order is same as supportedLanguages :+1: